packages/ec-core/src/operator/selector/weighted/mod.rs

```diff
@@ -0,0 +1,57 @@
+pub mod weighted;
+pub mod weighted_selector_pair;
+pub mod with_weighted_selector;
+
+#[derive(Debug, thiserror::Error)]
+#[error("Overflow while trying to calculate the sum of the weights {0} and {1}.")]
+pub struct WeightSumOverflow(u32, u32);
```

Comment:
I would either put this along the constructors (where the error occurs) or inside a seperate pub mod errors; - I would depend this on where we end up putting the other error.